src/beamlime/handlers/accumulators.py

```diff
@@ -117,34 +130,118 @@ def _cleanup(self) -> None:
+            coords={self._dim: sc.array(dims=['event'], values=pixel_ids, unit=None)},
+        )
+        self._chunks.clear()
+        return da.group(self._groups).fold(dim=self._dim, sizes=self._sizes)
+
+    def clear(self) -> None:
+        self._chunks.clear()
+
+
+class ROIBasedTOAHistogram(Accumulator[sc.DataArray, sc.DataArray]):
+    def __init__(self, config: Config, roi_filter: ROIFilter):
+        self._config = config
+        self._roi_filter = roi_filter
+        self._chunks: list[sc.DataArray] = []
+
+        # Note: Currently we are using the same ROI config values for all detector
+        # handlers ands views. This is for demo purposes and will be replaced by a more
+        # flexible configuration in the future.
+        self._roi_x = ConfigValueAccessor(
+            config=config,
+            key='roi_x',
+            default={'min': 0.0, 'max': 0.0},
+            convert=lambda raw: (raw['min'], raw['max']),
+        )
+        self._roi_y = ConfigValueAccessor(
+            config=config,
+            key='roi_y',
+            default={'min': 0.0, 'max': 0.0},
+            convert=lambda raw: (raw['min'], raw['max']),
+        )
+        self._nbin = -1
+        self._edges: sc.Variable | None = None
+        self._edges_ns: sc.Variable | None = None
+        self._current_roi = None
+
+    def _check_for_config_updates(self) -> None:
+        nbin = self._config.get('time_of_arrival_bins', 100)
+        if self._edges is None or nbin != self._nbin:
+            self._nbin = nbin
+            self._edges = sc.linspace(
+                'time_of_arrival', 0.0, 1000 / 14, num=nbin, unit='ms'
+            )
+            self._edges_ns = self._edges.to(unit='ns')
+            self.clear()
+
+        # Access to protected variables should hopefully be avoided by changing the
+        # config values to send indices instead of percentages, once we have per-view
+        # configuration.
+        y, x = self._roi_filter._indices.dims
+        sizes = self._roi_filter._indices.sizes
+        y_min, y_max = self._roi_y()
+        x_min, x_max = self._roi_x()
+        # Convert fraction to indices
+        y_indices = (int(y_min * (sizes[y] - 1)), int(y_max * (sizes[y] - 1)))
+        x_indices = (int(x_min * (sizes[x] - 1)), int(x_max * (sizes[x] - 1)))
+        new_roi = {y: y_indices, x: x_indices}
+
+        if new_roi != self._current_roi:
+            self._current_roi = new_roi
+            self._roi_filter.set_roi_from_intervals(sc.DataGroup(new_roi))
+            self.clear()  # Clear accumulated data when ROI changes
+
+    def _add_weights(self, data: sc.DataArray) -> None:
+        constituents = data.bins.constituents
+        content = constituents['data']
+        content.coords['time_of_arrival'] = content.data
+        content.data = sc.ones(
+            dims=content.dims, shape=content.shape, dtype='float32', unit='counts'
+        )
+        data.data = sc.bins(**constituents)
```

Comment:
Can the `_bins_no_validate` be used here for extra performance gain?

Reply:
Yes, thanks, fixed now!